// Copyright 2018 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

package android_webview;

option optimize_for = LITE_RUNTIME; // TODO(crbug/800281): Remove this after proto 4.0
option java_package = "org.chromium.android_webview.proto";

// WebView uses AwVariationsSeed to serialize a downloaded seed, along with the
// required HTTP header metadata, to a file. While Chrome stores the seed in
// prefs, WebView stores it in a separate file so the WebView service can write
// new seeds directly to each app's seed file.
//
// Next tag: 7
message AwVariationsSeed {
  // A base64-encoded signature of the seed data. This can be used by the client
  // to verify the seed contents. Comes from HTTP header "X-Seed-Signature".
  optional string signature = 1;
  // 2-letter country code. Comes from HTTP header "X-Country".
  optional string country = 2;
  // Date the seed was downloaded. Comes from HTTP header "Date"; see RFC 2616,
  // sections 3.3.1 and 14.18 for the format.
  // This was deprecated in favor of the |date| field in October 2019.
  optional string date_header = 3 [deprecated = true];
  // Whether seed_data is GZIP compressed. Comes from HTTP header "IM".
  optional bool is_gzip_compressed = 4;
  // The download body, itself a serialized VariationsSeed proto.
  optional bytes seed_data = 5;
  // Date the seed was downloaded according to the device's clock in
  // milliseconds since UNIX epoch, GMT.
  optional int64 date = 6;
}
